server/status/feed.go

```diff
@@ -40,6 +42,27 @@ type CallErrorEvent struct {
 	Method proto.Method
 }
 
+// TestSyncEvent is intended for testing only; it can be used by tests to
+// synchronize with feed consumers. A NodeEventListener will atomically
+// decrement the expected count, closing the Sync channel if the new count is 0.
+type TestSyncEvent struct {
+	// Sync is a read-only exported accessor for the same channel stored in 'close'.
+	Sync  <-chan struct{}
```

Comment:
Moved the entire `select{}`-based synchronization bit, with timeout, to a `Sync(timeout)` function on TestSyncEvent.